hydra/test_utils/test_utils.py

```diff
@@ -323,13 +335,18 @@ def experiment(cfg):
         OUTPUT_FILE=output_file,
         PROLOG=prolog_code,
     )
-    task_file = tmpdir / filename
+    task_file = modified_tmpdir / filename
     task_file.write_text(str(code), encoding="utf-8")
+
     cmd = [sys.executable, str(task_file)]
+    if task_config["generate_targets_file"]:
+        cmd = generate_custom_cmd(tmpdir, modified_tmpdir, filename)
+
```

Comment:
Please fix to either:
1. Do this inside the lambda you are passing.
2. Do this in a second lambda.